### PR TITLE
Update NXP Initial support Cadence MHDP8501(HDMI/DP) for i.MX8MQ patch and add support for Coral Edge TPU Development Board

### PR DESCRIPTION
**Pull request by heitbaum**
This PR adds support for the NXP `Google i.MX8MQ Phanbell` - https://coral.ai/products/dev-board/ 

The previous NXP HDMI was failing dcss startup due to the 27 mhz clock.

Current patchwork patch allows Phanbell to display and playback (no audio)

LE pastebin - https://paste.libreelec.tv/present-woodcock.log

- NXP (iMX8): add support for Coral Edge TPU Development Board
- linux (NXP iMX8): Initial support Cadence MHDP8501(HDMI/DP) for i.MX8MQ
- linux (NXP iMX8): update .config for 6.11

Future NXP work
- revisit atf
- revisit uboot
- pcie0 (QCA6174 802.11ac Wireless Network Adapter) - currently hangs during boot
- sd / emmc boot
- hdmi audio
- hantro-vpu 38310000.video-codec: dma alloc of size 5992448 failed
- cma: cma_alloc: reserved: alloc failed, req

### CODE REVIEW

**Comment by jernejsk** [comment]
> * hantro-vpu 38310000.video-codec: dma alloc of size 5992448 failed
> 
> * cma: cma_alloc: reserved: alloc failed, req

I imagine you either have to enable iommu (preferable, if possible) or increase CMA size, which should be in range of 384 MB or more (try 4k videos with different codecs)

**Comment by heitbaum** [comment]
> > * hantro-vpu 38310000.video-codec: dma alloc of size 5992448 failed
> > * cma: cma_alloc: reserved: alloc failed, req
> 
> I imagine you either have to enable iommu (preferable, if possible) or increase CMA size, which should be in range of 384 MB or more (try 4k videos with different codecs)

384MB works with HD x264. I had issues with 4K x265 - but will endeavor to look at iommu.

File: packages/linux-firmware/firmware-imx/package.mk [hunk 1]
```diff
@@ -20,11 +20,23 @@ unpack() {
 
 makeinstall_target() {
   mkdir -p ${INSTALL}/$(get_full_firmware_dir)/imx/sdma
-    cp -P firmware/sdma/sdma-imx6q.bin ${INSTALL}/$(get_full_firmware_dir)/imx/sdma
-    cp -P firmware/sdma/sdma-imx7d.bin ${INSTALL}/$(get_full_firmware_dir)/imx/sdma
+  case "${ARCH}" in
+    "aarch64")
+      cp -P firmware/sdma/sdma-imx7d.bin ${INSTALL}/$(get_full_firmware_dir)/imx/sdma
+      ;;
+    "arm")
```
[the remark this answers is not in the document]
Comment (heitbaum) [reply packages/linux-firmware/firmware-imx/package.mk:27]: The `PKG_ARCH="aarch64 arm"` fixes the adding of the missing blobs to the iMX8.

The case is there to “only include iMX6” for iMX6, and “iMX8” for iMX8. I probably should rewrite is for “DEVICE” not “ARCH”.

[the remark this answers is not in the document]
Comment (heitbaum) [reply packages/linux-firmware/firmware-imx/package.mk]: updated the makeinstall logic to only copy iMX6 firmware for iMX6 DEVICE, and iMX8 firmware for iMX8 DEVICE

**Comment by jernejsk** [comment]
gtg?

**Comment by heitbaum** [comment]
> gtg?

Yep.

**Review by jernejsk — APPROVED** [review]

**Comment by heitbaum** [comment]
> > * hantro-vpu 38310000.video-codec: dma alloc of size 5992448 failed
> > * cma: cma_alloc: reserved: alloc failed, req
> 
> I imagine you either have to enable iommu (preferable, if possible) or increase CMA size, which should be in range of 384 MB or more (try 4k videos with different codecs)

Looks like SMMU is a no go on the I.MX8MQ
https://patchew.org/Xen/1559938596-5696-1-git-send-email-amittomer25@gmail.com/
```
There are some important differences between the
iMX8M and that SoC beyond those you've noted (no A72 cluster, no SMMU, and no
System Control Unit (SCU)), but it might solve some of the issues you've been
running into or avoid yet unknown issues when you attempt to boot a guest domain.
```

**Pull request merged**